kimchi/src/circuits/polynomials/range_check/witness.rs

```diff
@@ -79,152 +24,118 @@ impl ZeroWitnessCell {
 ///     For example, we can convert the `RangeCheck0` circuit gate into
 ///     a 64-bit lookup by adding two copy constraints to constrain
 ///     columns 1 and 2 to zero.
-pub const WITNESS_SHAPE: [[WitnessCell; COLUMNS]; 4] = [
-    /* row 1, RangeCheck0 row */
-    range_check_row(0),
-    /* row 2, RangeCheck0 row */
-    range_check_row(1),
-    /* row 3, RangeCheck1 row */
-    [
-        ValueWitnessCell::create(),
-        /* 2-bit crumbs (placed here to keep lookup pattern */
-        /*               the same as RangeCheck0) */
-        LimbWitnessCell::create(2, 0, 86, 88),
-        LimbWitnessCell::create(2, 0, 84, 86),
-        /* 12-bit plookups */
-        LimbWitnessCell::create(2, 0, 72, 84),
-        LimbWitnessCell::create(2, 0, 60, 72),
-        LimbWitnessCell::create(2, 0, 48, 60),
-        LimbWitnessCell::create(2, 0, 36, 48),
-        /* 2-bit crumbs */
-        LimbWitnessCell::create(2, 0, 34, 36),
-        LimbWitnessCell::create(2, 0, 32, 34),
-        LimbWitnessCell::create(2, 0, 30, 32),
-        LimbWitnessCell::create(2, 0, 28, 30),
-        LimbWitnessCell::create(2, 0, 26, 28),
-        LimbWitnessCell::create(2, 0, 24, 26),
-        LimbWitnessCell::create(2, 0, 22, 24),
-        LimbWitnessCell::create(2, 0, 20, 22),
-    ],
-    /* row 4, Zero row */
+fn layout<F: PrimeField>() -> [[Box<dyn WitnessCell<F>>; COLUMNS]; 4] {
     [
-        ZeroWitnessCell::create(),
-        /* 2-bit crumbs (placed here to keep lookup pattern */
-        /*               the same as RangeCheck0) */
-        LimbWitnessCell::create(2, 0, 18, 20),
-        LimbWitnessCell::create(2, 0, 16, 18),
-        /* 12-bit plookups (see note about copies in range_check_row) */
-        CopyWitnessCell::create(0, 1),
-        CopyWitnessCell::create(0, 2),
-        CopyWitnessCell::create(1, 1),
-        CopyWitnessCell::create(1, 2),
-        /* 2-bit crumbs */
-        LimbWitnessCell::create(2, 0, 14, 16),
-        LimbWitnessCell::create(2, 0, 12, 14),
-        LimbWitnessCell::create(2, 0, 10, 12),
-        LimbWitnessCell::create(2, 0, 8, 10),
-        LimbWitnessCell::create(2, 0, 6, 8),
-        LimbWitnessCell::create(2, 0, 4, 6),
-        LimbWitnessCell::create(2, 0, 2, 4),
-        LimbWitnessCell::create(2, 0, 0, 2),
-    ],
-];
+        /* row 1, RangeCheck0 row */
+        range_check_0_row("v0", 0),
```

Comment:
This seems a tad brittle, depending on the variable name `v0` used elsewhere.

Reply:
There's also a way to map them explicitly with `variable_map!["v0" => blar_dee_blar]`, if that helps.